enhancements/update/cluster-version-operator-x509-upstream-trust.md

```diff
@@ -0,0 +1,202 @@
+	//      -----END CERTIFICATE-----
+	//
+	// +optional
+	TrustedCA *ConfigMapNameReference `json:"trustedCA,omitempty"`
+}
+```
+
+[`ClusterVersionSpec`][api-spec] will be extended to include the following new property:
+
+```go
+// upstreamTLS specifies settings for TLS connections to the upstream
+// update service.  If trustedCA is set, it is used directly and not
+// merged with the system default trust bundle.  If upstreamTLS is set,
+// even if none of its constituent properties are set, it takes
+// precedence over the proxy configuration.  To use the proxy
+// configuration for connections to the upstream update service, unset
+// upstreamTLS completely.
+// +optional
+UpstreamTLS *TLSConfig `json:"upstreamTLS,omitempty"
+```
+
+### User Stories
+
+#### Local update recommendation service with non-standard certificate authority
+
+Users running a local [update recommendation service][update-service] (such as [Cincinnati][cincinnati]) may use a serving certificate signed by an internal CA.
+This enhancement will allow the CVO to successfully pull recommended updates from those services.
+Connecting to the services over HTTP (without TLS) would be unwise, because an attacker who can man-in-the-middle the update recommendation service can suggest clusters do crazy things (e.g. "I see you're on 4.4.3.  I suggest you update to 4.1.0"), for which there are no CVO-side guards at the moment.
+And in some cases (e.g. updating to fumbled releases which we initially expected to be healthy), no CVO-side guards are possible.
+This enhancement enables trusted, secure connections to the update recommendation service without requiring users to provision a certificate signed by a default CA.
+
+#### Transparent, network-level proxies
+
+[rhbz#1773419][rhbz1773419] discusses issue when CVO upstream requests are not redirected via an `httpsProxy` setting, but are instead routed through a proxy via network-level configuration (e.g. all outgoing TCP traffic is routed through a proxy).
+The CVO currently only injects the Proxy config's `trustedCA` data [when `httpsProxy` is non-empty][trustedCA-vs-httpsProxy], so there is currently no mechanism for configuring the CVO's trusted CA store in this situation, short of forcing an explicit value into `httpsProxy`.
+While it's possible that the Proxy semantics would encourage consumers to ingest `trustedCA` in the absence of `httpsProxy`, this enhancement allows for users to configure trusted CAs without going through the Proxy object.
+
+### Implementation Details/Notes/Constraints
+
+The network operator supports copying trust bundles between ConfigMaps based on labels on the target ConfigMaps.
+But because we are not using the Proxy configuration, the CVO will need to grow its own implementation to retrieve data from the referenced ConfigMap in the `openshift-config` namespace.
+Because it is the CVO retrieving and consuming the data, there is no need to copy a ConfigMap over to volume-mount into the CVO pod.
```

Comment:
need more details on how you'll do this.  you need to copy the CA content over before you initialize your TLS transport that will be talking to the service in question.  (not too hard, just want to make sure that's accounted for).

Reply:
> ... you need to copy the CA content over before you initialize your TLS transport that will be talking to the service in question...

CVO does this today, using the ConfigMap referenced from Proxy's `trustedCA` (when `httpsProxy` is set).  We'd just have to twiddle the existing logic slightly to slot the new properties in at higher precedence.  Added links to that code to the enhancement with 47f96bb -> c5ba370.